### PR TITLE
CHT Issue 18. #1765

### PR DESCRIPTION
**Pull request by kogot**
Hey guys please check that PR. @Aprogiena @noescape00 
Thanks

### CODE REVIEW

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs [hunk 1]
```diff
@@ -2046,6 +2046,47 @@ public void ChainTHasCheckpointAt1000_MaxReorgIs10_TwoChainsPriorTo1000Presented
...
+            ctx.SetupPeersForTest(cht, initialChainTip);
+
+            // Additional SetUp for current test.
+            ChainedHeader chainATip = cht.GetPeerTipChainedHeaderByPeerId(0);
+            ChainedHeader[] last2HeadersPeerA = {
+                chainATip.Previous, chainATip
+            };
+            var intitialChainedHeaders = new Dictionary<uint256, ChainedHeader>(cht.GetChainedHeadersByHash());
+
+            // Checking that 8a,9a in chained tree.
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: 8a,` `9a `are` in

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs [hunk 2]
```diff
@@ -2046,6 +2046,47 @@ public void ChainTHasCheckpointAt1000_MaxReorgIs10_TwoChainsPriorTo1000Presented
...
+            ChainedHeader chainDTip = cht.GetPeerTipChainedHeaderByPeerId(3);
+
+            // Peed A claims Peer D chain.
+            chainATip = chainDTip;
+
+            List<BlockHeader> peerABlockHeaders = ctx.ChainedHeaderToList(chainATip, chainATip.Height);
+
+            cht.ConnectNewHeaders(0, peerABlockHeaders);
+
+            var chainedHeadersAfterPeerAChainged = new Dictionary<uint256, ChainedHeader>(cht.GetChainedHeadersByHash());
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: Changed

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs [hunk 3]
```diff
@@ -2046,6 +2046,47 @@ public void ChainTHasCheckpointAt1000_MaxReorgIs10_TwoChainsPriorTo1000Presented
...
+
+            // Checking that 8a,9a not presented in chained tree anymore.
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: 8a,` `9a, `are` not

**Comment by Aprogiena** [comment]
Looks good, up to @noescape00

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs (hunk 3 above)
Comment (kogot) [reply src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: will fix, thanks

**Pull request merged**